src/output-json-verdict.c

```diff
@@ -0,0 +1,228 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Juliana Fajardini <jufajardini@oisf.net>
+ *
+ * JSON Verdict log module to log the matched rules' final verdict on the packet
+ * information
+ *
+ */
+
+#include "action-globals.h"
+
+#include "output.h"
+#include "output-json.h"
+#include "output-json-alert.h"
+#include "output-json-verdict.h"
+
+#include "packet.h"
+
+#include "suricata.h"
+#include "suricata-common.h"
+
+#define MODULE_NAME "JsonVerdictLog"
+
+typedef struct JsonVerdictOutputCtx_ {
+    uint8_t flags;
+    OutputJsonCtx *eve_ctx;
+} JsonVerdictOutputCtx;
+
+typedef struct JsonVerdictLogThread_ {
+    JsonVerdictOutputCtx *verdict_ctx;
+    OutputJsonThreadCtx *ctx;
+} JsonVerdictLogThread;
+
+void GetVerdictJsonInfo(JsonBuilder *jb, const Packet *p)
+{
+    jb_open_object(jb, "verdict");
+    /* add verdict info */
+    if (PacketCheckAction(p, ACTION_DROP) && EngineModeIsIPS()) {
+        JB_SET_STRING(jb, "action", "drop");
+    } else if (PacketCheckAction(p, ACTION_REJECT_ANY)) {
+        // check rule to define type of reject packet sent
+        if (EngineModeIsIPS()) {
+            JB_SET_STRING(jb, "action", "drop");
+        } else {
+            JB_SET_STRING(jb, "action", "alert");
+        }
+        if (p->action & ACTION_REJECT || p->action) {
+            JB_SET_STRING(jb, "reject-target", "source");
+        } else if (p->action & ACTION_REJECT_DST) {
+            JB_SET_STRING(jb, "reject-target", "destination");
+        } else if (p->action & ACTION_REJECT_BOTH) {
+            JB_SET_STRING(jb, "reject-target", "both");
+        }
+        JB_SET_STRING(jb, "reject", "[tcp-reset, icmp-prohib, user defined]");
```

Comment:
This is the bit I wasn't sure about the approach, @victorjulien .  The examples from https://github.com/OISF/suricata/pull/8596#discussion_r1197787688 show all options. If that's the case, would this work? Or do we want to check what is actually being done by Suri, and output only that option? If so, I'm done sure how to figure out when will it be `user defined`... 

The other two I figured I could decide by checking the packet proto.

Reply:
this shouldn't be a single string, but a json array of strings

Reply:
"user defined" can be ignored, it was a place holder for some future ideas

Reply:
I see, thanks.